autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/CredentialListSorter.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.ui.credential.management
+
+import com.duckduckgo.autofill.AutofillDomainFormatter
+import com.duckduckgo.autofill.domain.app.LoginCredentials
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface CredentialListSorter {
+    fun sort(credentials: List<LoginCredentials>): List<LoginCredentials>
+}
+
+@ContributesBinding(AppScope::class)
+class CredentialListSorterByTitleAndDomain @Inject constructor(
+    private val domainFormatter: AutofillDomainFormatter
+) : CredentialListSorter {
+
+    override fun sort(credentials: List<LoginCredentials>): List<LoginCredentials> {
+        return credentials.sortedWith(
+            compareBy(
+                titleSort(),
+                domainSort()
+            )
+        )
+    }
+
+    private fun titleSort(): (LoginCredentials) -> Comparable<*>? = { it.domainTitle }
```

Comment:
How will this handle special characters vs numbers?

Reply:
Special characters and numbers better handled in the latest commits